lib/kafka_producer.js

```diff
@@ -202,8 +203,16 @@ KafkaProducer.prototype.batch = function batch(topic, message, timeStamp, callba
         return;
     }
 
+    if (typeof message !== 'string' && !Buffer.isBuffer(message)) {
+        var err = new Error('For batching, message must be a string or buffer, not ' + (typeof message));
+        if (!callback) {
+            throw err;
```

Comment:
This lib is also used as logging part, most people will just fire and forget.
I don't think we should fail application because of this.
You can add some switch to turn on throw exception if necessary. Also, most people who care about msg delivery will passing callback in. 
Default behavior should be msg dropped.